src/firebase-db.ts

```diff
@@ -494,3 +497,82 @@ export const getLegacyLinkedInteractiveInfo = (embeddableRefId: string, laraData
       });
     });
 };
+
+export const getApRun = async (sequenceActivity?: string|null) => {
+  if (!portalData) {
+    throw new Error("Must set portal data first");
+  }
+  let query: firebase.firestore.Query = app.firestore().collection(apRunsPath());
+
+  if (portalData.type === "authenticated") {     // logged in user
+    query = query
+      .where("platform_id", "==", portalData.platformId)
+      .where("resource_url", "==", portalData.resourceUrl)
+      .where("context_id", "==", portalData.contextId)
+      .where("platform_user_id", "==", portalData.platformUserId.toString());
+  } else {
+    query = query.where("run_key", "==", portalData.runKey);
+  }
+
+  // for sequence runs the sequence_activity will be set but will be null for activity only runs
+  if (sequenceActivity) {
+    query = query.where("sequence_activity", "==", sequenceActivity);
+  }
+
+  const doc = await query.get();
+  if (doc.empty) {
+    return null;
+  }
+
+  return {id: doc.docs[0].id, data: doc.docs[0].data() as IApRun};
```

Comment:
I don't see any sorting here. It seems to just be using the first document that it finds. So if if the activities in the sequence are visited in order lets say 1, 2, 3. Then the user goes back to activity 2. And then they quit. Now if they launch the sequence again, I'd expect it won't open activity 2. I tested it with this URL: 
https://activity-player.concord.org/branch/save-page-position/?sequence=sample-sequence
And as expected it opens activity 1 instead of activity 2.

Can you add a test for this kind of navigation and reloading? It seems like that part of the code isn't covered by the existing tests in this PR.